### PR TITLE
Load root component

### PR DESCRIPTION
**Pull request by TheMcMurder**

### CODE REVIEW

File: src/single-spa-vue.js [hunk 1]
```diff
@@ -47,6 +51,10 @@ function mount(opts, mountedInstances, props) {
         otherOptions.el = props.domElement;
       }
 
+      if (!opts.appOptions.render && !opts.appOptions.template && opts.rootComponent) {
+        otherOptions.render = (h) => h(opts.rootComponent)
+      }
+
```
Comment (TheMcMurder) [inline src/single-spa-vue.js]: In order for this condition to be satisfied you have to have the following configuration:
```js
const vueLifecycles = singleSpaVue({
  Vue,
  loadRootComponent: () => import(/* webpackChunkName: "planets-app" */'./App.vue').then(property('default')),
  appOptions: {
    el: '#planets',
    router: configuredRouter,
  }
})
```

File: src/single-spa-vue.js [hunk 2]
```diff
@@ -47,6 +51,10 @@ function mount(opts, mountedInstances, props) {
         otherOptions.el = props.domElement;
       }
 
+      if (!opts.appOptions.render && !opts.appOptions.template && opts.rootComponent) {
+        otherOptions.render = (h) => h(opts.rootComponent)
```
Comment (joeldenning) [inline src/single-spa-vue.js]: Oh I didn't even realize this was an option for rendering Vue apps. I sorta like this way better than the template way.

Let's update the documentation to encourage people down this route instead, with loadRootComponent or rootComponent?

Comment (joeldenning) [reply src/single-spa-vue.js]: Also I think we should change the documentation from exporting arrays for the lifecycles to just exporting the one function -- no need to make it more complicated than it has to be, i think

File: README.md [hunk 3]
```diff
@@ -43,6 +44,7 @@ All options are passed to single-spa-vue via the `opts` parameter when calling `
 
 - `Vue`: (required) The main Vue object, which is generally either exposed onto the window or is available via `require('vue')` `import Vue from 'vue'`.
 - `appOptions`: (required) An object which will be used to instantiate your Vue.js application. `appOptions` will pass directly through to `new Vue(appOptions)`
+- `loadRootComponent`: (optional and replaces `appOptions.render`) A promise that resolves with your root component. This is useful for lazy loading.
```
Comment (joeldenning) [inline README.md]: The `render` opt should be documented, too

Comment (joeldenning) [reply README.md]: actually nevermind -- i misunderstood this

File: src/single-spa-vue.js (hunk 2 above)
Comment (EvanBurbidge) [reply src/single-spa-vue.js]: @joeldenning if you want to see some good examples of this dig into the vuetify git repository.  great work on this lads, you saved me so much heart ache :D

**Pull request merged**